app/controllers/settings/ldap_test_controller.rb

```diff
@@ -0,0 +1,136 @@
+require "net/ldap"
+require "json"
+require "openssl"
+require "base64"
+
+# LdapTestController is used to validate LDAP Connector before saving
+class Settings::LdapTestController < ApplicationController
+  def ldap_filter_search(ldap, baseDN, filter_string)
+    filter = Net::LDAP::Filter.construct(filter_string)
+    attrs = ["mail", "cn", "sn", "objectclass"]
+    search_object = ldap.search(
+      base:          baseDN,
+      filter:        filter,
+      attributes:    attrs,
+      return_result: true
+    )
+    search_object
```

Comment:
I believe this function could just end in ldap.search, and the return of it would automatically be returned by the function. As is it does make it more clear that it is intended to return search_object though, so it may be better to stay this way.

Reply:
Fixed now.